### ISSUE TITLE
ql.filter is ignored

### ISSUE THREAD

**Issue by trib0r3** [qilingframework/qiling#648]
**Describe the bug**
`ql.filter` seems to be ignoring assigned filters:

```txt
$ python3 solve.py
[=] [unistd.py:189]     brk(0x0)
[=] [utsname.py:25]     uname(0x80000000d9f0) = 0
[=] [unistd.py:219]     access(/etc/ld.so.nohwcap, 0x0) = -1
[=] [unistd.py:219]     access(/etc/ld.so.preload, 0x4) = -1
[=] [fcntl.py:84]       openat(4294967196, /etc/ld.so.cache, 0x0, 0o0) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64-linux-gnu/tls/i686/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64-linux-gnu/tls/i686/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64-linux-gnu/tls/i686/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64-linux-gnu/tls/i686, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64-linux-gnu/tls/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64-linux-gnu/tls/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64-linux-gnu/tls/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64-linux-gnu/tls, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64-linux-gnu/i686/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64-linux-gnu/i686/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64-linux-gnu/i686/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64-linux-gnu/i686, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64-linux-gnu/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64-linux-gnu/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64-linux-gnu/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64-linux-gnu, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /usr/lib/x86_64-linux-gnu/tls/i686/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/usr/lib/x86_64-linux-gnu/tls/i686/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /usr/lib/x86_64-linux-gnu/tls/i686/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/usr/lib/x86_64-linux-gnu/tls/i686, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /usr/lib/x86_64-linux-gnu/tls/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/usr/lib/x86_64-linux-gnu/tls/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /usr/lib/x86_64-linux-gnu/tls/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/usr/lib/x86_64-linux-gnu/tls, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /usr/lib/x86_64-linux-gnu/i686/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/usr/lib/x86_64-linux-gnu/i686/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /usr/lib/x86_64-linux-gnu/i686/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/usr/lib/x86_64-linux-gnu/i686, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /usr/lib/x86_64-linux-gnu/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/usr/lib/x86_64-linux-gnu/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /usr/lib/x86_64-linux-gnu/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/usr/lib/x86_64-linux-gnu, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/tls/i686/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/tls/i686/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/tls/i686/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/tls/i686, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/tls/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/tls/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/tls/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/tls, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/i686/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/i686/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/i686/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/i686, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/x86_64/libc.so.6, 0x0, 0o0) = -1
[=] [stat.py:456]       stat(/lib/x86_64, 0x80000000d440) = -1
[=] [fcntl.py:84]       openat(4294967196, /lib/libc.so.6, 0x0, 0o0) = 3
[=] [unistd.py:267]     read(3, 0x80000000d5a8, 0x340) = 832
[=] [stat.py:322]       fstat(3, 0x80000000d440) = 0
[=] [mman.py:200]       mmap(0x0, 0x3f0ae0, 0x5, 0x802, 3, 0) = 0x7fffb7dd6000
[=] [mman.py:54]        mprotect(0x7fffb7fbd000, 0x200000, 0x0) = 0
[=] [mman.py:200]       mmap(0x7fffb81bd000, 0x6000, 0x3, 0x812, 3, 1994752) = 0x7fffb81bd000
[=] [mman.py:200]       mmap(0x7fffb81c3000, 0x3ae0, 0x3, 0x32, 4294967295, 0) = 0x7fffb81c3000
[=] [unistd.py:234]     close(3) = 0
[=] [mman.py:200]       mmap(0x0, 0x2000, 0x3, 0x22, 4294967295, 0) = 0x7fffb81c7000
[=] [prctl.py:19]       arch_prctl(0x7fffb81c7f00) = 0
[=] [mman.py:54]        mprotect(0x7fffb81bd000, 0x4000, 0x1) = 0
[=] [mman.py:54]        mprotect(0x55555575b000, 0x1000, 0x1) = 0
[=] [mman.py:54]        mprotect(0x7ffff7ffc000, 0x1000, 0x1) = 0
[=] [unistd.py:189]     brk(0x0)
[=] [unistd.py:189]     brk(0x555555780000)
[=] [stat.py:322]       fstat(1, 0x80000000dc00) = 0
[=] [ioctl.py:105]      ioctl(0x1, 0x5401, 0x80000000db60) = -1
[=] [unistd.py:286]     write(1,55555575f2b0,5) = 0
test
[=] [unistd.py:234]     close(1) = 0
```

**Sample Code**
```python
def main():
    rootfs = path.join(environ["QILING_ROOTFS"], "x8664_linux")
    ql = Qiling(["./echo", "test"], rootfs=rootfs)
    ql.filter = ["mmap"]

    ql.run()

if __name__ == "__main__":
    main()
```

**Expected behavior**
qiling will trace only of `mmap` function

**Comment by xwings** [comment]
we are doing a factoring and part of this being "ignored" when move from nprint to logging.

we will fix it once the refactoring is done

**Comment by wtdcode** [comment]
Fixed in 09476f70c3065fbb3c7d6b58f6fd3c8e0428a4ed

**Comment by xwings** [comment]
fixed and close
